lib/std/start.zig

```diff
@@ -492,44 +503,47 @@ fn callMainWithArgs(argc: usize, argv: [*][*:0]u8, envp: [][*:0]u8) u8 {
     return initEventLoopAndCallMain();
 }
 
-fn main(c_argc: c_int, c_argv: [*][*:0]c_char, c_envp: [*:null]?[*:0]c_char) callconv(.C) c_int {
+fn cMain(c_argc: c_int, c_argv: [*][*:0]c_char, c_envp: [*:null]?[*:0]c_char) callconv(.C) c_int {
     var env_count: usize = 0;
     while (c_envp[env_count] != null) : (env_count += 1) {}
     const envp = @ptrCast([*][*:0]u8, c_envp)[0..env_count];
 
-    if (builtin.os.tag == .linux) {
+    if (native_os == .linux) {
         const at_phdr = std.c.getauxval(elf.AT_PHDR);
         const at_phnum = std.c.getauxval(elf.AT_PHNUM);
         const phdrs = (@intToPtr([*]elf.Phdr, at_phdr))[0..at_phnum];
-        expandStackSize(phdrs);
+        linuxExpandStackSize(phdrs);
     }
 
-    return @call(.always_inline, callMainWithArgs, .{ @intCast(usize, c_argc), @ptrCast([*][*:0]u8, c_argv), envp });
+    return @call(.always_inline, posixCallMainWithArgs, .{ @intCast(usize, c_argc), @ptrCast([*][*:0]u8, c_argv), envp });
 }
 
-fn mainWithoutEnv(c_argc: c_int, c_argv: [*][*:0]c_char) callconv(.C) c_int {
+fn cMainWithoutEnv(c_argc: c_int, c_argv: [*][*:0]c_char) callconv(.C) c_int {
     std.os.argv = @ptrCast([*][*:0]u8, c_argv)[0..@intCast(usize, c_argc)];
     return @call(.always_inline, callMain, .{});
 }
 
-// General error message for a malformed return type
-const bad_main_ret = "expected return type of main to be 'void', '!void', 'noreturn', 'u8', or '!u8'";
+/// Error message for a malformed main return type.
+const bad_main_ret = if (std.os.ExitStatus == void)
+    "expected return type of main to be 'void', '!void, or 'noreturn'"
+else
+    std.fmt.comptimePrint("expected return type of main to be 'void', '!void, '{[ty]s}', '!{[ty]s}', or 'noreturn'", .{ .ty = @typeName(std.os.ExitStatus) });
```

Comment:
Or
```suggestion
    "expected return type of main to be 'void', '!void, 'std.os.ExitProcess', '!std.os.ExitProcess', or 'noreturn'";
```
which would encourage users to actually write that type for their main return types which in turn would be better for cross-platform.
Wouldn't that be better? It also solves the problem of ugly/confusing strings returned by `@typeName` (as it is the case on the UEFI, read "Regression:" in https://github.com/ziglang/zig/pull/16094/commits/6787dc8d04c29bca8a871e1c31adcef27a1e4d73).